packages/blocks/table/src/components/Header.tsx

```diff
@@ -38,9 +40,17 @@ export const Header: VFC<HeaderProps> = ({
           <Menu label="Filter">
             <FilterDetail
               columns={columns}
-              onFilter={(filters) =>
-                onAggregate({ operation: "filter", multiFilter: filters })
-              }
+              onFilter={(filters) => {
+                if (!entityTypeId) {
+                  return;
+                }
+
+                onAggregate({
+                  operation: "filter",
+                  multiFilter: filters,
+                  entityTypeId,
+                });
```

Comment:
There has been a general recommendation in my previous PRs to exit early in case one's running some code in a conditional. Hence, I'd like to keep it as it is for now.

Reply:
This was probably me - I think it's easier to read code if there's some early exit condition that's one can take, rather than nesting big blocks of code in a conditional.

@benwerner01 why do you think this worth changing?

Reply:
I don't think it's worth changing if not every developer feels the same way, as this is purely a subjective preference one way or the other. Also want to be clear this is non-blocking.

The reason I avoid early returns when I write code is so that it is more clear to the reader which scope depends on what conditionals to run.

I've come across complex functions (particularly in react code), where there's multiple conditionals that result in returns at different stages of the function, which in my opinion makes it harder to tell whilst reading which part of the code in the same scope will run when. So as a general rule I try to be explicit about what code runs when by using a conditional that creates a new scope, and try to avoid introducing early returns.

A common argument for early returns is that it avoids having lots of nested conditionals, but I'd argue if the function's logic is complex enough for nested conditionals it likely justifies creating more than one function to encapsulate the various functionality.

But again, don't want to impose my subjective preferences on other people if that's not their experience, so if you disagree @akash-joshi don't worry about making the change.

Reply:
I think this is the sort of thing which should be flagged when it's actually affecting the readability of the code in question - I can see your point of view and think it would largely depend on what the conditionals looked like as to which approach would make the code more readable. 

I don't think it makes any difference to this example, and therefore is probably not worth spending time raising and discussing in a review - I'm guilty of contributing to this but wanted to understand why it was raised. Thank you for explaining!